match/lib/match/utils.rb

```diff
@@ -58,6 +58,16 @@ def self.get_cert_info(cer_certificate_path)
       return {}
     end
 
+    def self.check_cert_validity(cer_certificate_path)
+      cert = OpenSSL::X509::Certificate.new(File.binread(cer_certificate_path))
+      now = Time.now.utc
+      if (cert.not_before <=> now) * (now <=> cert.not_after) < 1
```

Comment:
yes, no problem :)

line 62: i get the certificate from the provided path (i use OpenSSL lib which do this perfectly)
line 63: i get the present time, in UTC format because start and end time of certificate are in UTC format (Apple's choice)
line 64: the tricky part :D 
_date1 <=> date2  = -1_ if date1 < date2
_date1 <=> date2  = 0_ if date1 = date2
_date1 <=> date2  = 1_ if date1 > date2

so if if (cert.not_before <=> now) * (now <=> cert.not_after) < 1
it means it is equal to 0 or -1

if 0, it means start OR end date is now, it means actually end date is now because start date is the date of creation of certificate. so certificate is invalid

if -1, it means
Option A: start <=> now = -1 AND now <=> end = 1 which means start < now AND end < now which means certificate is out of date
Option B: start <=> now = 1 AND now <=> end = -1 which means now < start AND now < end it is impossible because start < end

Reply:
Ohh, one of those. I can never remember these. Thanks!

Reply:
Jumping in here as another Ruby noob, can we update the code to use some less cool Ruby here (and therefore probably use more lines of code), so that it's easier for contributors in the future to read and update the code? :)